### PR TITLE
CAN 2.0B support

### PR DESCRIPTION
**Pull request by JasonBrave**
Note I did not update canlib git submodule as https://github.com/waterloo-rocketry/canlib/pull/69 have not been merged

### CODE REVIEW

File: user_config.c [hunk 1]
```diff
@@ -53,15 +53,15 @@ void parse_usb_string(const char *input) {
...
                 // --> each data value contains 1-2 nibbles
                 case CHECK_DATA:
                     if (char_to_check == ',' || char_to_check == ';') {
                         data_index = 2;
                         if (!sid_sent) {
                             sid_sent = true;
-                            msg.sid = (int)strtol(data_to_send, &nullptr, 16);
-                            data_to_send[4] = 0;
+                            msg.sid = strtol(data_to_send, &nullptr, 16);
+                            memset(data_to_send + 4, 0, 6);
```
Comment (Rio6) [inline user_config.c:64]: where does this 6 come from?

File: user_config.c [hunk 2]
```diff
@@ -74,7 +74,7 @@ void parse_usb_string(const char *input) {
                             break;
                         }
 
-                    } else if (!sid_sent && data_index < 5) {
+                    } else if (!sid_sent && data_index < 10) {
                         data_to_send[data_index] = char_to_check;
                         data_index++;
                         break;
```
Comment (Rio6) [inline user_config.c:80]: does this condition and the one below it do the same thing? Also I'd use `sizeof(data_to_send) - 1` (if that's what 10 is here).

File: user_config.c [hunk 3]
```diff
@@ -124,9 +124,7 @@ void parse_usb_string(const char *input) {
                                 mcp_can_send(&msg);
                                 num_data_sent = 0;
                                 sid_sent = false;
-                                data_to_send[2] = 0;
-                                data_to_send[3] = 0;
-                                data_to_send[4] = 0;
+                                memset(data_to_send + 2, 0, 8);
```
Comment (Rio6) [inline user_config.c:127]: where does this 8 come from?

**Review by Rio6 — APPROVED** [review]
all nits, if it's tested and working then lgtm

**Review by StarlightDescender — APPROVED** [review]
LGTM :)

**Comment by JasonBrave** [comment]
Will address Rio's comment in a future PR

**Pull request merged**